sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/intervalExpressions.scala

```diff
@@ -137,7 +137,11 @@ abstract class IntervalNumOperation(
       val iu = IntervalUtils.getClass.getName.stripSuffix("$")
       s"""
         try {
-          ${ev.value} = $iu.$operationName($interval, $num);
+          if ($num != 0) {
```

Comment:
This check should be added to interpreted code as well: https://github.com/apache/spark/pull/26410/files#diff-a1b6eef1170a6b4595b16af2c366be53R129

Reply:
Actually, I tested this. I guess the literals already go through this, but they are correct to produce nulls with zero. I have no idea, adding check will do no harm

Reply:
How did you test? Could you add a test to `IntervalExpressionsSuite` and use `checkEvaluation()` which checks with and without codegen.

Reply:
The literals will do `eval` not `codegen`